src/main/java/org/mule/service/http/impl/service/util/RedirectUtils.java

```diff
@@ -51,35 +59,55 @@ public static boolean isRedirected(int statusCode) {
   }
 
   /**
-   * @param requestMethod
    * @param responseStatusCode
-   * @return the original request method if the status code is 301, 307 or 308, or GET if it is 302 or 303.
+   * @return false if the status code is 301, 307 or 308, or true if it is 302 or 303.
    */
-  public static String getMethodForStatusCode(String requestMethod, int responseStatusCode) {
+  public static boolean sendAsGet(int responseStatusCode) {
     switch (responseStatusCode) {
       case 301:
       case 307:
       case 308:
-        return requestMethod;
+        return false;
       case 302:
       case 303:
-        return HttpConstants.Method.GET.name();
+        return true;
       default:
         throw new IllegalArgumentException("Invalid status code");
     }
   }
 
   /**
-   * Create a new request with the params of the original and the new URI from the LOCATION header.
+   * Create a new request with the params of the original and the new URI from the LOCATION header. This method is copy from
+   * AhcEventFilter#RedirectHandler.newRequest. The set-cookie header is handle in GrizzlyHttpClient.createGrizzlyRedirectRequest
    * 
    * @param response HttpResponse
    * @param request  HttpRequest
    * @return an HttpRequest request.
    */
-  public static HttpRequest createRedirectRequest(HttpResponse response, HttpRequest request) {
+  public static HttpRequest createRedirectRequest(HttpResponse response, HttpRequest request, HttpRequestOptions options) {
     Uri path = create(create(request.getUri().toString()), response.getHeaders().get(LOCATION));
-    return builder().uri(path.toUrl()).method(getMethodForStatusCode(request.getMethod(), response.getStatusCode()))
-        .protocol(request.getProtocol()).headers(request.getHeaders()).entity(request.getEntity())
-        .build();
+
+    MultiMap<String, String> headers = new MultiMap<>(request.getHeaders());
+    headers.remove(Host.toString());
+    headers.remove(ContentLength.toString());
+    String redirectMethod;
+
+    if (sendAsGet(response.getStatusCode())) {
```

Comment:
Add a comment and encapsulate in a method indicating the rationale behind this switch to redirect method.

Reply:
I'm trying to make this method as similar as possible to the original (https://github.com/mulesoft/grizzly-ahc/blob/support/1_14-mule/src/main/java/com/ning/http/client/providers/grizzly/AhcEventFilter.java#L799).